### PR TITLE
Fixing lockfile drift

### PR DESCRIPTION
**Pull request by trevoring-okta**
## Description:
- Adds `yarn_sync()` in `setup.sh` to check for changes to `yarn.lock` in CI and force users to check in an updated lockfile
- Removes `@okta/okta-signin-widget: "link:../../dist"` in test app `package.json` because it was leading to an empty resolution in the lockfile on every `yarn install`, which would raise a false positive for the `yarn_sync()` logic
- Ensures that `yarn install` in `setup` uses public yarn registry and then reverts to internal artifactory for `@okta:registry` for all following scripts
- Cleans up naming of different registries used for `setup`, `publish`, and `verify-registry-install` scripts

## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [x] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-653621](https://oktainc.atlassian.net/browse/OKTA-653621)
- [Bacon Link](https://bacon-go.aue1e.saasure.net/commits?artifact=okta-signin-widget&branch=ti-OKTA-653621-fix-lockfile-drift&page=1&pageSize=12&sha=8faa60bdc0f49018cb90c18f75af49d3a3e516bc&tab=main)

### CODE REVIEW

File: test/app/package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "test": "echo \"Error: no test specified\" && exit 1"
   },
   "dependencies": {
-    "@okta/okta-signin-widget": "link:../../dist"
```
Comment (shawyu-okta) [inline test/app/package.json:12]: @denysoblohin-okta you had originally added this for reasons related to the ESM build for the test app. @trevoring-okta and I couldn't find any problems running the ESM build now with this change, so maybe it's not longer a problem now. Can you take a look and explain if there's any cases that you think will fail?

File: scripts/verify-registry-install.sh [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 # NOTE: MUST BE RAN *AFTER* THE PUBLISH SUITE
 
-export REGISTRY="${ARTIFACTORY_URL}/npm-topic"
+export PUBLISHED_REGISTRY="${ARTIFACTORY_URL}/npm-topic"
```
Comment (shawyu-okta) [inline scripts/verify-registry-install.sh:5]: Should this be `PUBLISH_REGISTRY` to match the one in `publish.sh` with the same value?

File: test/app/package.json (hunk 1 above)
Comment (trevoring-okta) [reply test/app/package.json:12]: @shawyu-okta Sorry forgot to update you directly via Slack DM but I already checked with @denysoblohin-okta and he verified that the `link:../../dist` is no longer necessary

File: scripts/setup.sh [hunk 3]
```diff
@@ -20,13 +28,23 @@ if ! setup_service yarn 1.22.19 &> /dev/null; then
   exit ${FAILED_SETUP}
 fi
 
-cd ${OKTA_HOME}/${REPO}
+cd "${OKTA_HOME}"/"${REPO}" || exit
```
Comment (denysoblohin-okta) [inline scripts/setup.sh:31]: Just curious, why  this change is needed?

Comment (trevoring-okta) [reply scripts/setup.sh:31]: Ah that was leftover from copying some logic from a different script, it's not necessary I can change it back to minimize changes

File: test/app/package.json (hunk 1 above)
Comment (shawyu-okta) [reply test/app/package.json:12]: Ok nice, no worries

**Review by shawyu-okta — APPROVED** [review]
lgtm. Did you run a test to make sure the `yarn_sync` is catching things as you'd expect in CI?

**Comment by trevoring-okta** [comment]
@shawyu-okta Yeah before removing the `link:../../dist`, the `yarn_sync()` was causing failures on Bacon for `setup.sh` after `yarn install` because the lockfile was getting changed to have the empty resolution

**Pull request merged**